CBinding/Project/CProject.cs

```diff
@@ -90,118 +92,135 @@ public class CProject : Project, IDeployable
 		/// Extensions for C/C++ header files
 		/// </summary>
 		public static string[] HeaderExtensions = { ".H", ".HH", ".HPP", ".HXX" };
-		
-		private void Init ()
-		{
-			packages.Project = this;
 
-			if (IdeApp.IsInitialized)
-				IdeApp.Workspace.ItemAddedToSolution += OnEntryAddedToCombine;
-		}
-		
-		public CProject ()
+		public Dictionary<string, bool> BOMPresentInFile = new Dictionary<string, bool>();
```

Comment:
Avoid public fields as it breaks encapsulation. A method `public bool IsBomPresentInFile (string filename)` would be better.